[9.0][MIG] product_variant_configurator

ThomasBinsfeld opened the pull request:
- [ ] #38 

Closed. See : #44

oihane committed:
[MOD] New name and security files fixed

oihane committed:
[MOD] <product_variants_no_automatic_creation> _product_find

oihane committed:
[MOD] Fix pep8

oihane committed:
[MOD] possible values added to scheduled lines for attributes

oihane committed:
[MOD] Attributes in mo product lines

oihane committed:
[MOD] Operation assignation fixing

oihane committed:
[MOD] Fixed errors and changes

oihane committed:
[MOD] translations

oihane committed:
[MOD] get inherit attribute value, and corresponding product if exists

oihane committed:
[MOD] <product_variants_no_automatic_creation> prices according to attributes

oihane committed:
[MOD] Fixing pricelist aplication

oihane committed:
[FIX] PEP8

oihane committed:
[MOD] import missing

oihane committed:
[FIX] <product_variants_no_automatic_creation> view don't allow editing

oihane committed:
Update product_view.xml

Commit:
Add module product_variant_csv_import

Commit:
Add attribute_line_ids

pedrobaeza committed:
[IMP] product_variants_no_automatic_creation: Enhance labels and defaults

pedrobaeza committed:
[FIX] product_variants_no_automatic_creation: Labels

bealdav committed:
[IMP] icon product variant

oihane committed:
[MOD] variants creation fixed

oihane committed:
[MOD] Avoid launching onchange with default value

oihane committed:
[MOD] Sale_variants description changed, and uom in purchase when template selected

pedrobaeza committed:
[IMP] Expand AUTHORS

pedrobaeza committed:
[ADD] product_supplierinfo_variant

Product suppplier info per variant, not per template
====================================================

This module limits supplier info to product variant scope, instead of having
the same record for all variants.

Installation
============

When you install the module, a copy of each supplier info record is made for
each variant, so the process can take a long time.

pedrobaeza committed:
[FIX/IMP] product_variants_no_automatic_creation: Preserve order on attributes + improve code

pedrobaeza committed:
[FIX] product_variants_no_automatic_creation: Make product template search usable without variants. Closes #804

pedrobaeza committed:
[FIX] product_variants_no_automatic_creation: Adapt _product_find for looking for non full-set of values

Commit:
[ADD] product_variant_cost: nuevo modulo que añade campo coste para la variante de producto

Commit:
[IMP] product_variant_cost: cambiar nombre de función charge -> load

oihane committed:
[IMP] Some view changes and translation added

sysadminmatmoz committed:
Missing templates and translations added

sysadminmatmoz committed:
Missing templates and translations added

pedrobaeza committed:
[IMP] product_variants_no_automatic_creation: Include constraint for product/attribute uniqueness

pedrobaeza committed:
OCA Transbot updated translations from Transifex

pedrobaeza committed:
OCA Transbot updated translations from Transifex

oihane committed:
[IMP] product_variant_cost_price: migrated to OCA/product-variant from odoomrp-wip

oihane committed:
[IMP] product_variant_cost_price: removed some code, updated post-init

eLBati committed:
ADD README, IMP __openerp__.py

pedrobaeza committed:
[FIX] product_variants_no_automatic_creation: Find correctly a product for its attributes

Adding to the domain all values that exists, not only the ones that have more than one
possible value. That was an old constraint because attributes with only one value were
not populated, but that was fixed a long time ago and now it makes that the match doesn't
happen because the comparison of the attributes length fail.

pedrobaeza committed:
OCA Transbot updated translations from Transifex

pedrobaeza committed:
[IMP] product_variants_no_automatic_creation: Add configurator in product variant form

Commit:
[IMP] Tests added

pedrobaeza committed:
[FIX] product_variants_no_automatic_creation: Security for product.product.attribute

pedrobaeza committed:
[FIX] product_variants_no_automatic_creation: Don't error on missing attribute values

pedrobaeza committed:
[IMP] product_variant_cost: Tests, template<>product changes, README...

pedrobaeza committed:
Merge pull request #14 from odoomrp/8.0-variant_cost

[8.0][ADD] New module product_variant_cost_price

OCA-git-bot committed:
[UPD] addons table in README.md

pedrobaeza committed:
[IMP] product_variants_no_automatic_creation: Avoid to disturb tests results

pedrobaeza committed:
[FIX] product_variant_cost_price: Author

pedrobaeza committed:
[FIX] product_variant_cost: post_init_hook - no duplicate ir.property

pedrobaeza committed:
[FIX/IMP] README version+badges

rvalyi committed:
Merge pull request #15 from eLBati/add_product_variant_csv_import

Add product variant csv import

OCA-git-bot committed:
[UPD] addons table in README.md

pedrobaeza committed:
[FIX] product_variants_no_automatic_creation: Fix security file

jcarlosmontoya committed:
Update term es.po

tafaRU committed:
Rename module according to the nomenclature used in this repo till now

tafaRU committed:
Adapt README to OCA template

tafaRU committed:
Modify files accoding to OCA template

sergio-teruel committed:
[ADD][8.0] product_variant_supplierinfo: finished module to OCA

sergio-teruel committed:
[IMP][8.0] product_variant_supplierinfo: More test coverage

OCA-git-bot committed:
[ADD] setup.py

tafaRU committed:
Merge pull request #2 from sergio-teruel/add-product_supplierinfo_variant

[IMP][8.0] product_variant_supplierinfo: More test coverage

pedrobaeza committed:
[IMP] product_variant_*: Abstract product configurator + application on stock.move

sergio-teruel committed:
[ADD][8.0] product_variant_sale_price: New module

hugosantosred committed:
Fix _check_configuration_validity constraint

oihane committed:
[FIX+IMP] product_variants_no_automatic_creation:

* Fix expected singleton error when updating attributes (#1237)
* Tests

oihane committed:
[IMP] product_variant_cost_price: when template has only one variant,
template standard_price is written twice, creating two entries in history

oihane committed:
[IMP] product_variant_cost_price: added test for template standard_price edition

sergio-teruel committed:
[NEW][8.0] product_variant_weight: New module

sergio-teruel committed:
[NEW][8.0] product_variant_weight: Changes suggested by @pedrobaeza

sergio-teruel committed:
[FIX][8.0] product_variant_weight: Fix flake8

oihane committed:
[IMP] product_variant_cost_price: fixed errors

oihane committed:
[IMP] product_variant_cost_price: updated tests

oihane committed:
[IMP] product_variant_cost_price: when adding new value to attribute_lines
creates 2 history lines

pedrobaeza committed:
Merge pull request #27 from oihane/8.0-cost_price

[IMP] product_variant_cost_price: when template has only one variant,

oihane committed:
[IMP] add transifex to travis file

pedrobaeza committed:
Merge pull request #29 from oihane/8.0-transifex

[IMP] add transifex to travis file

sergio-teruel committed:
[NEW][8.0] product_variant_weight: Changes suggested by @yajo.

oca-transbot committed:
OCA Transbot updated translations from Transifex

pedrobaeza committed:
Merge pull request #28 from Tecnativa/8.0-PR-product_variant_weight

[NEW][8.0] product_variant_weight. Allow store weights at product.product level

OCA-git-bot committed:
[UPD] addons table in README.md

OCA-git-bot committed:
[ADD] setup.py

oca-transbot committed:
OCA Transbot updated translations from Transifex

tafaRU committed:
Apply some PEP8 cleanup

tafaRU committed:
Avoid redundant loop

Commit:
[IMP] product_variant_cost_price: synchronize cost method when product and template relation is one2one

pedrobaeza committed:
Merge pull request #31 from agaldona/8.0-variant_cost_price

[IMP] product_variant_cost_price: synchronize cost method when product and template relation is one2one

sebastienbeau committed:
[REF] refactor module in order to be able to have generic supplier based on the product template or specific supplier for a variant.

pedrobaeza committed:
[IMP] product_variants_no_automatic_creation

* Link quick-created attribute value to the template
* Don't require parent product_tmpl_id field in view

sebastienbeau committed:
[REF] simplify code with pedro comment

sebastienbeau committed:
[IMP] update readme and author

sebastienbeau committed:
[REF] refactor doc string

sebastienbeau committed:
Merge pull request #32 from akretion/8.0-refactor-product_supplierinfo_variant

8.0 refactor product supplierinfo variant

OCA-git-bot committed:
[UPD] addons table in README.md

OCA-git-bot committed:
[ADD] setup.py

oca-transbot committed:
OCA Transbot updated translations from Transifex

sbidoul committed:
[FIX] remove en.po that was erroneously created by transbot

sergio-teruel committed:
[NEW][8.0] product_variant_sale_price: Add test, fixes and translation

sergio-teruel committed:
[NEW][8.0] product_variant_sale_price: Improvements

sergio-teruel committed:
[NEW][8.0] product_variant_sale_price: Fix travis

tafaRU committed:
[FIX][8.0] product_variant_sale_price: fix E121 continuation line under-indented for hanging indent

tafaRU committed:
Add product_variant_available_in_pos module

tafaRU committed:
[FIX][8.0] product_variant_sale_price: fix copyright and README according to OCA template

tafaRU committed:
[FIX][8.0] product_variant_sale_price: fix post_init_hook (set_sale_price_on_variant), manage case in wich template_is ids None

pedrobaeza committed:
Merge pull request #30 from tafaRU/8.0-product_variant_sale_price-add

[8.0] Add product_variant_sale_price module

OCA-git-bot committed:
[UPD] addons table in README.md

OCA-git-bot committed:
[ADD] setup.py

oca-transbot committed:
OCA Transbot updated translations from Transifex

tafaRU committed:
[product_variant_sale_price] Also set fix_price to variant when template is created

tafaRU committed:
[product_variant_sale_price] Set minimun fix_price among variants to ensure consistency with the price shown in the shop

gurneyalex committed:
Merge pull request #37 from tafaRU/8.0-product_variant_sale_price_fix

[8.0] [FIX] product_variant_sale_price

Garamotte commented on product_variant_configurator/models/product_template.py:
```diff
@@ -0,0 +1,95 @@
+    def onchange_no_create_variants(self):
+        if not self._origin:
+            return {}
+        return {'warning': {'title': _('Change warning!'),
```
You may test the new value, and display only if the value is not "yes" (which always blocks variants creation).

Garamotte commented on product_variant_configurator/models/product_category.py:
```diff
@@ -0,0 +1,34 @@
+    def onchange_no_create_variants(self):
+        if not self._origin:
+            return {}
+        return {'warning': {'title': _('Change warning!'),
```
You may test the new value, and display the warning only if the value is False.

Garamotte commented on product_variant_configurator/models/product_configurator.py:
```diff
@@ -0,0 +1,237 @@
+                    'owner_model': self._name,
+                    'owner_id': self.id,
+                })
+            self.product_attribute_ids = [(0, 0, x) for x in attribute_list]
```
You can directly append the tuples `(0, 0, data)` in `attribute_list` and remove the for loop.

Garamotte commented on product_variant_configurator/models/product_configurator.py:
```diff
@@ -0,0 +1,237 @@
+        if not self.product_id:
+            product_tmpl = self.product_tmpl_id
+            values = self.product_attribute_ids.mapped('value_id')
+            if self._fields.get('partner_id'):
```
You should use `if 'partner_id' in self._fields` instead of the `get` method.

Garamotte commented on product_variant_configurator/models/product_configurator.py:
```diff
@@ -0,0 +1,237 @@
+                obj = self.env['product.attribute.value'].with_context(
+                    lang=self.partner_id.lang)
+                values = obj.browse(
+                    self.product_attribute_ids.mapped('value_id').ids)
```
You should use `values.ids`, instead of calling the `mapped` method twice.

Garamotte commented on product_variant_configurator/models/product_attribute_value.py:
```diff
@@ -0,0 +1,25 @@
+            template = self.env['product.template'].browse(
+                self.env.context['template_for_attribute_value'])
+            line = template.attribute_line_ids.filtered(
+                lambda x: x.attribute_id == attr_value.attribute_id)
```
Instead of using browse + filtered, you can use :

```
line = self.env['product.attriute.line'].search([
    ('product_tmpl_id', '=', self.env.context['template_for_attribute_value']),
    ('attribute_id', '=', attr_value.attribute_id.id),
])
```

Garamotte commented on product_variant_configurator/models/product_configurator.py:
```diff
@@ -0,0 +1,237 @@
+                for attribute_id in op[2]:
+                    attribute = attribute_obj.browse(attribute_id)
+                    if attribute.value_id:
+                        value_ids.append(attribute.value_id.id)
```
Maybe replace by something like :

```
attribute_values = attribute_obj.browse(op[2]).mapped('value_id')
value_ids.extend(attribute_values.ids)
```

Garamotte commented on product_variant_configurator/models/product_pricelist.py:
```diff
@@ -0,0 +1,159 @@
+            return {}
+
+        version = False
+        for v in pricelist.version_id:
```
Pricelist versions have been removed on v9.0.

Garamotte commented on product_variant_configurator/models/product_configurator.py:
```diff
@@ -0,0 +1,237 @@
+    product_attribute_ids = fields.One2many(
+        comodel_name='product.configurator.attribute',
+        domain=lambda self: [("owner_model", "=", self._name)],
+        inverse_name='owner_id', string='Product attributes', copy=True)
```
Using an Integer field as inverse_name of a One2many field crashes when deleting records from the One2many, because the ORM searches for an `ondelete` attribute on the field, which only is on Many2one fields.

Garamotte reviewed (changes requested):
Some comments are applicable to the 8.0 version, but not all, that's why I commented here.

oca-transbot committed:
OCA Transbot updated translations from Transifex

Garamotte commented:
Also, you shouldn't copy/paste the module in a big commit, but instead base this PR on #38 (using this guide https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-9.0#technical-method-to-migrate-a-module-from-80-to-90-branch), to keep commits history.

pedrobaeza committed:
Merge pull request #34 from tafaRU/8.0-product_variant_available_in_pos-add

[8.0][ADD] product_variant_available_in_pos module

OCA-git-bot committed:
[ADD] setup.py

OCA-git-bot committed:
[UPD] addons table in README.md

zakmak88 committed:
[ADD] product_variant_configurator

This module has been ported from the odoomrp repo.
the module name in the odoo mrp repo was product_variants_no_automatic_creation
the following improvement where done while porting this module:

* Improve the readme file with a technical description on how the module should
  be used with other modules.
* Add attribute col=4 to group parent for a better desplay of product category
  for view.
* Remove unnecessary @api.multi before compute or onchange methods.
* Move product.configurator.attribute model to the new
  product_configurator_attribute.py file.
* Add unit test for product.configurator.attribute model that includes test for
  _compute_possible_value_ids and _compute_price_extra methods.
* Rename XML files in views and security folder as per OCA guidelines.
* Improve test coverage for product_configurator.
* Add unit test for purchase.order and purchase.order.line models.
* Separate the purchase.order and purchase.order.line models code into two files.
* Rename XML files as per OCA guidelines.
* product_variant_configurator: set product_tmpl_id in
  onchange_product_id_product_configurator_old_api and
  onchange_product_id_product_configurator methods for product.configurator model.
* Add unit test for onchange_product_tmpl_id and onchange_product_attribute_ids.

bealdav commented on product_variant_configurator/README.rst:
```diff
@@ -0,0 +1,87 @@
+
+To configure the creation of the variants behaviour, you need to:
+
+#. Go to ``Warehouse > Products``, and select a product.
```
No dependency on 'stock'. Warehouse menu not available.

zakmak88 committed:
[FIX] product_variant configurator: Fix _price_rule_get_multi

Fix _price_rule_get_multi method inspired from v10 implementation
so the migration will be easier to v9 and v10

zakmak88 committed:
[FIX] product_variant configurator: Unit test for  _price_rule_get_multi

zakmak88 committed:
[FIX] product_variant_configurator: ad sequence to pricelist test

zakmak88 committed:
[MIG] product_variant_configurator: Adapt _price_rule_get_multi to v9 version

zakmak88 committed:
[FIX]product_variant_configurator: Remove duplicated xml files

zakmak88 committed:
[MIG] product_variant_configurator: Remove group tag from category view

the category view in v9 doesn't have group with name parent

zakmak88 committed:
[MIG] product_variant_configurator: Adapt product template view to v9 version

zakmak88 committed:
[MIG] product_variant_configurator: Adapt pricelist test to v9 version

zakmak88 committed:
[MIG] product_variant_configurator: Adapt test_product_variant_configurator to v9

zakmak88 committed:
[MIG] product_variant_configurator: Adapt product.product form view to v9

zakmak88 committed:
[MIG] product_variant_configurator: Add condition when product_tmpl isn't set

zakmak88 committed:
[MIG] product_variant_configurator: Adapt _price_get_multi to new api

zakmak88 committed:
[MIG] product_variant_configurator: Adapt pricelist test example to v9
